public/js/modal.js

```diff
@@ -0,0 +1,37 @@
+// ommitted parameters won't be changed
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function toggleModal(newAction, newTitle, newDesc, newButtonText, newIsActionFunction = 0) {
+  const title = document.getElementById('modal-title')
+  const description = document.getElementById('modal-desc')
+  const submitButton = document.getElementById('submit-button')
+  const funcButton = document.getElementById('func-button')
+  const form = document.getElementById('modal-form')
+
+  if (newTitle) {
+    title.innerHTML = newTitle
+  }
+  if (newDesc) {
+    description.innerHTML = newDesc
+  }
+  if (newButtonText) {
+    submitButton.innerHTML = newButtonText
+    funcButton.innerHTML = newButtonText
+  }
+  if (newAction) {
+    form.action = newAction
+    funcButton.setAttribute( 'onClick', `javascript: ${newAction}`)
+  }
+
+  if (newIsActionFunction !== 0) {
+    if (newIsActionFunction) {
+      form.classList.add('hidden')
+      funcButton.classList.remove('hidden')
+    } else {
+      funcButton.classList.add('hidden')
+      form.classList.remove('hidden')
+    }
+  }
```

Comment:
Meh, zero. `undefined` could mean more and then the if down below in line 25 would be:

```js
if (typeof newIsActionFunction !== 'undefined') {
```

Reply:
Also, maybe the function's parameters could be in an object like `params: {newAction, newTitle, newDesc, newButtonText, newIsActionFunction}`

This way undefined doesn't have to be entered when calling the function

But then also, you couldn't use conditionals like `if (newDesc)`, but you would have to write them as `if (typeof params.newDesc !== 'undefined')`. :(